### DISCUSSION TITLE
How does the tick system function?

### DISCUSSION THREAD

**Discussion by LouChiSoft** [RavEngine/RavEngine#16] [Q&A] [answered]
I get the impression that the CPU side of the code is decoupled from the FPS of the application (this might be wrong). But setting `SetMinTickTime` doesn't seem to have an effect on my application. I set it to a ridiculously high value of 1000 milliseconds and I expected it to update the world once per second but it ran just as fast as before and `GetApp()->CurrentTPS()` returns anywhere from 200-300.

I'm hesitant to call it a bug because I could just as easily be getting the intended use case wrong.

For more context, when I had the min tick rate set to 1 millisecond I would have expected the application to tick 1000 times a second and have a scale of 1. And then if it drops to say 500 ticks per second because I loaded up the systems it would be a frame scale of 2. But currently no matter what I set it too I get a tick rate around 200-300 and a frame scale of less then 1.

**Comment by Ravbug** [comment, marked as answer]
`SetMinTickTime` currently does nothing. In a (much earlier) version of the engine, the tick rate and the frame rate were indeed independent. However, I changed this because it added a lot of complexity and overhead for relatively little benefit. This function is a vestige of that earlier design. In the current version of the engine, the tick rate is the same as the frame rate. 

Apologies for the confusion. I will remove `SetMinTickTime` at some point.

**Reply by LouChiSoft** [reply]
Thanks for the explaination. Disappointing that it's not decoupled though. I thought thta was a neat feature of the engine and was hoping to use it. Can I assume then that the frameScale has also now changed to a more standard delta time?

**Reply by Ravbug** [reply]
Framescale is still based on 60fps = 1. So if your game runs at 30fps, framescale will be 2, and at 120fps it will be 0.5.
